### PR TITLE
[BugFix] Return error if open HDFS file failed

### PR DESCRIPTION
**Pull request by letian-jiang**
Fixes #issue

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: be/src/exec/pipeline/sink/iceberg_table_sink_operator.cpp [hunk 1]
```diff
@@ -131,12 +131,13 @@ Status IcebergTableSinkOperator::push_chunk(RuntimeState* state, const ChunkPtr&
...
-            _partition_writers[partition_location]->append_chunk(chunk.get(), state);
+            return _partition_writers[partition_location]->append_chunk(chunk.get(), state);
         } else {
-            partition_writer->second->append_chunk(chunk.get(), state);
+            return partition_writer->second->append_chunk(chunk.get(), state);
         }
     }
-    return Status::OK();
+
+    CHECK(false) << "unreachable";
```
Comment (stephen-shelby) [inline be/src/exec/pipeline/sink/iceberg_table_sink_operator.cpp:140]: please add some more precious message

Comment (letian-jiang) [reply be/src/exec/pipeline/sink/iceberg_table_sink_operator.cpp:140]: literally unreachable since all branches have been returned

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by stephen-shelby — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**